File: src/main/java/at/searles/regexp/RepCount.java
package at.searles.regexp;

class RepCount implements Regexp {
    private final Regexp regexp;
    private final int count;

    RepCount(Regexp regexp, int count) {
        this.regexp = regexp;
        this.count = count;
    }

    @Override
    public <A> A accept(Visitor<A> visitor) {
        return visitor.visitRepCount(regexp, count);
    }

    @Override
    public String toString() {
        return String.format("count(%s, %d)", regexp, count);
    }
}

File: src/main/java/at/searles/regexp/RepCount.kt
package at.searles.regexp

internal class RepCount(private val regexp: Regexp, private val count: Int) : Regexp {
    override fun <A> accept(visitor: Visitor<A>): A {
        return visitor.visitRepCount(regexp, count)
    }

    override fun toString(): String {
        return String.format("count(%s, %d)", regexp, count)
    }

}